src/utils/walletConnect.ts

```diff
@@ -172,35 +261,35 @@ async function rejectProposal({
 export async function pair({ uri }: { uri: string }) {
   logger.debug(`WC v2: pair`, { uri }, logger.DebugContext.walletconnect);
 
-  try {
-    // show loading state as feedback for user
-    Navigation.handleAction(Routes.WALLET_CONNECT_APPROVAL_SHEET, {});
-
-    const { topic } = parseUri(uri);
-    const client = await signClient;
-
-    await client.core.pairing.pair({ uri });
-
-    const timeout = setTimeout(() => {
-      showErrorSheet();
-      analytics.track('New WalletConnect session time out');
-    }, 10_000);
-
-    const handler = (
-      proposal: SignClientTypes.EventArguments['session_proposal']
-    ) => {
-      // listen for THIS topic pairing, and clear timeout if received
-      if (proposal.params.pairingTopic === topic) {
-        client.off('session_proposal', handler);
-        clearTimeout(timeout);
-      }
-    };
+  /**
+   * Make sure this is cleared if we get multiple pairings in rapid succession
+   */
+  if (PAIRING_TIMEOUT) clearTimeout(PAIRING_TIMEOUT);
```

Comment:
Fixed a race condition here where we had multiple timeouts live at the same time.